pandas/tests/indexing/test_indexing.py

```diff
@@ -231,6 +231,7 @@ def test_setitem_dtype_upcast(self):
 
         df.loc[0, "c"] = "foo"
         expected = DataFrame([{"a": 1, "c": "foo"}, {"a": 3, "b": 2, "c": np.nan}])
+        expected = expected[df.columns]  # compensate dict key ordering
         tm.assert_frame_equal(df, expected)
```

Comment:
you can pass an argument to assert_frame_equal to ignore the column order (and then I would also only do that for older pythons, and ensure that on recent python actually the correct `expected` is created)

Reply:
didn't know that. will do.

Reply:
Can you do the check_like trick also here?

Reply:
done